src/docs/develop/variables.md

```diff
@@ -33,13 +34,92 @@ alt="Screenshot of Raw Editor"
 layout="responsive"
 width={1954} height={1303} quality={100} />
 
-Variables can be defined as simple key/value pairs or as [Templated Variables](#templated-variables) (eg. `${{Postgres.DATABASE_URL}}`),
-which can dynamically reference other variables, shared variables, or plugin variables (more on this below).
+## Shared Variables
+
+Shared variables help reduce duplication of variables across multiple services within the same project.  They are defined at the project environment level and can be added in Project Settings > Shared Variables.
+
+<Image src="https://res.cloudinary.com/railway/image/upload/v1669678393/docs/shared-variables-settings_vchmzn.png"
+alt="Screenshot of Shared Variables Settings"
+layout="responsive"
+width={2402} height={1388} quality={100} />
+
+To use a shared variable, either click the Share button and select the desired services,
+or visit the Variables tab within the service itself and click "Shared Variable".
+
+<Image src="https://res.cloudinary.com/railway/image/upload/v1667332192/docs/shared-variables-picker_ryjble.png"
+alt="Screenshot of Shared Variables Picker"
+layout="responsive"
+width={1784} height={1168} quality={100} />
+
+Adding a shared variables to a service creates a [Reference Variable](/develop/variables#reference-variables) in the service.
+
+## Reference Variables
+
+Variables can be defined to reference variables in other services or to reference shared variables.  This is useful for ease of maintenace and security, allowing you to set a variable in a single place, wherever it makes the most sense.
+
+Reference variables use Railway's [template syntax](/develop/variables#template-syntax) using a specific format - `${{NAMESPACE.VAR}}` - where `NAMESPACE` can either be a service's name or the value `shared` (depending on where the variable is defined that you need to access) and `VAR` is the variable name.
+
+When using reference variables, you also have access to [Railway-provided variables](/develop/variables#railway-provided-variables).
+
+### Example Scenarios
 
-### Multiline Variables
+Here are some example scenarios to help clarify reference variable usage and syntax.
+
+#### Referencing a Shared variable -
+- You have a shared variable defined in your project called `API_KEY`, and you need to make the API key available to a service.  Go to the service's variables tab, and add a variable with the following value:
+  - `API_KEY=${{shared.API_KEY}}`
```

Comment:
So, since variable references are required for shared variables, should we introduce references before them?

Here, you have to double back again

Reply:
Tried it different ways, I think I like Shared before Reference because you also see Shared in Reference so it's circular.  but this way, as soon as you get to the end of the Shared section that talks about how it creates a reference, you go right into Reference.